Fix R2RDump in the presence of composite images

trylek opened the pull request:
In a recent change DavidWr modified the Crossgen2 manifest metadata
emitter by adjusting the assembly index offset - before his change,
manifest assembly indices started at 1, now they start at 2 as 1
newly represents the manifest metadata itself. This change fixes the
counterpart assembly index calculations in R2RDump.

Thanks

Tomas

/cc @dotnet/crossgen-contrib

trylek committed:
Fix R2RDump in the presence of composite images

In a recent change DavidWr modified the Crossgen2 manifest metadata
emitter by adjusting the assembly index offset - before his change,
manifest assembly indices started at 1, now they start at 2 as 1
newly represents the manifest metadata itself. This change fixes the
counterpart assembly index calculations in R2RDump.

Thanks

Tomas

davidwrighton commented on src/coreclr/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs:
```diff
@@ -756,7 +756,7 @@ private void ParseMethodDefEntrypoints(Action<ReadyToRunSection, IAssemblyMetada
                     if (ReadyToRunAssemblyHeaders[assemblyIndex].Sections.TryGetValue(ReadyToRunSectionType.MethodDefEntryPoints, out methodEntryPointSection))
                     {
-                        methodDefSectionReader(methodEntryPointSection, OpenReferenceAssembly(assemblyIndex + 1));
+                        methodDefSectionReader(methodEntryPointSection, OpenReferenceAssembly(assemblyIndex + 2));
```
You need to make this change under a check for the R2R version. The latest version of R2RDump needs to be able to read old files.

trylek replied:
Thanks David for pointing that out, fixed in 2nd commit.

davidwrighton reviewed (changes requested):
Add the R2R version check, and I'll approve this.

trylek committed:
Address David's PR feedback

I have added the version check and a property representing the
component assembly start index.

Thanks

Tomas
